# **Introduction**

## **Chi-test (boroughs + svi)**
## **Bar-chart with svi as regression/scatterplot (zipcode)**

source: https://www.atsdr.cdc.gov/place-health/media/pdfs/2024/10/SVI2022Documezipcodetion.pdf

source: https://www.atsdr.cdc.gov/place-health/php/svi/svi-interactive-map.html

Chi-test: Using the average_year_eviction_zipcode_count for Chi-square test for zipcode, we get results as following:

1) SVI: P-value is 0.0000 and chi-square is 31.25. We can reject the null hypothesis. The Chi-square suggests a strong-to-moderate deviation from the distribution if the two groups/categories are independent. The odds are 83.31, meaning if you live in a zipcode with higher than average svi, you are 83.31 times more likely to be evicted than living in a zipcode with less than average svi.

2) Black + Hispanic Population percezipcodege: P-value is 0.0000 and chi-squiare is 72.40. We can refute the null hypothesis. We can say we are confident that high "black+hispanic percetage" boroughs are more likely to have high evictions (or vice versa) or there is a strong association between high BH pct and high eviction rates in the full dataset. The odds are 33.49, meaning if you live in a zipcode with b+h population above the average pct of nyc, you are 12.5125 times more likely to get evicted than someone living in a zipcode with below average percezipcodege of black + hispanic people.

In [ ]:
# !pip install geopandas folium matplotlib seaborn scipy esda splot

In [159]:
import pandas as pd
import numpy as np
from scipy import stats
import datetime as dt
import matplotlib
import matplotlib.pyplot as plt
import os
import io
import geopandas as gpd
import seaborn as sns
from scipy.stats import chi2_contingency
# suppress warning
import warnings
warnings.filterwarnings('ignore')

%matplotlib inline

In [51]:
pd.set_option('display.float_format', lambda x: '%.4f' % x)
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

# **Step 1 Get the Eviction data**

In [52]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [91]:
# data source:
file_path1 = '/content/drive/My Drive/X999/bbl_evictions_311_svi_normal_times_svi_zipcode.csv'
file_path2 = '/content/drive/My Drive/X999/bbl_evictions_311_svi_covid_svi_zipcode.csv'

In [92]:
evictions_pre_post_raw = pd.read_csv(file_path1)
evictions_covid_raw = pd.read_csv(file_path2)
evictions_covid_raw.shape, evictions_pre_post_raw.shape

((5386, 120), (66395, 121))

In [93]:
evictions_pre_post_raw.isna().sum().sum()

np.int64(0)

In [94]:
nan_rows = evictions_pre_post_raw[evictions_pre_post_raw.isna().any(axis=1)]  #
nan_rows[['zipcode']]

Empty DataFrame
Columns: [zipcode]
Index: []

In [95]:
evictions_pre_post = evictions_pre_post_raw.copy()
evictions_covid = evictions_covid_raw.copy()

In [96]:
evictions_pre_post.head(2)

primary_key         bbl court_index_number  docket_number  \
0  *308072/22_5865  3037420029         *308072/22           5865   
1  *313639/23_5202  3057940012         *313639/23           5202   

  eviction_address eviction_apartment_number executed_date   borough  zipcode  \
0   356 MILLER AVE            1 AND BASEMENT    2024-12-04  BROOKLYN    11207   
1  710 61ST STREET                 2ND FLOOR    2024-03-04  BROOKLYN    11220   

          ejectment eviction/legal_possession  latitude  longitude  \
0  Not an Ejectment                Possession   40.6721   -73.8911   
1  Not an Ejectment                Possession   40.6359   -74.0119   

   community_board  council_district  census_tract      bin               nta  \
0           5.0000           37.0000     1152.0000  3083989     East New York   
1           7.0000           38.0000      118.0000  3143881  Sunset Park East   

   year month_year                      geometry  yearbuilt bldgclass  \
0  2024    2024-12  POINT (-73.891105 40.672121)  1930.0000        C0   
1  2024    2024-03  POINT (-74.011883 40.635941)  1920.0000        B2   

   numfloors  unitsres            ownername  bldgarea building_type  \
0     3.0000    3.0000       356 MILLER LLC 2700.0000       pre-war   
1     2.0000    2.0000  A.R.M. PARKING, LLC 1204.0000       pre-war   

  building_category  is_condo floor_category                    rent_era  \
0           walk-up     False       low-rise  Pre-1947, pre-rent-control   
1        two-family     False       low-rise  Pre-1947, pre-rent-control   

               architectural_style                       economic_period  \
0  1921–1930, Art Deco Skyscrapers  1930-1945, great depression and WWII   
1            1900–1920, Beaux-Arts        Pre-1929, pre-great depression   

  residential_units_category  is_llc building_size_category  \
0                  3-5 units    True                  small   
1                     2-unit    True             very small   

       size_quartile     decade   fips   e_totpop  rpl_theme1  rpl_theme2  \
0        Q3 (50-75%)  1930-1939  11207 96801.0000      0.9788      0.9140   
1  Q1 (smallest 25%)  1920-1929  11220 93008.0000      0.9885      0.7635   

   rpl_theme3  rpl_theme4  rpl_themes  ep_pov150  ep_unemp  ep_nohsdp  \
0      0.9808      0.9812      0.9839    33.9000   11.1000    19.1000   
1      0.9594      0.9179      0.9662    37.5000    7.5000    37.9000   

   ep_uninsur  ep_age65  ep_age17  ep_disabl  ep_limeng  ep_noveh  ep_crowd  \
0      6.0000   13.8000   22.5000    13.8000     5.3000   57.8000    9.1000   
1     11.6000   13.1000   25.4000     8.4000    40.2000   61.7000   23.7000   

   ep_hburd  ep_afam  ep_hisp  ep_asian  ep_aian  ep_nhpi  ep_twomore  \
0   44.7000  55.9000  32.8000    1.5000   0.0000   0.0000      2.9000   
1   43.6000   1.7000  40.9000   40.7000   0.4000   0.0000      1.2000   

   ep_otherrace  ep_minrty  ep_white  invalid_zip svi_quartile    svi_group  \
0        1.6000    94.7000    5.3000        False           Q3  medium-high   
1        0.2000    85.0000   15.0000        False           Q3  medium-high   

   air_quality  animal_issues  appliances  building_exterior  doors_windows  \
0       0.0000         0.0000      1.0000             0.0000         1.0000   
1       0.0000         0.0000      0.0000             0.0000         0.0000   

   electrical_issues  elevator_issues  floors_stairs  general_complaints  \
0             2.0000           0.0000         0.0000              1.0000   
1             0.0000           0.0000         0.0000              0.0000   

   graffiti_posting  heat_hot_water  homeless_issues  noise_complaints  \
0            0.0000          3.0000           0.0000            0.0000   
1            0.0000          1.0000           0.0000            0.0000   

   other_issues  pest_issues  plumbing_issues  police_matters  \
0        0.0000       0.0000           3.0000          0.0000   
1        0.0000       0.0000           1.0000        

In [97]:
evictions_covid.head(2)

primary_key         bbl court_index_number  docket_number  \
0   004123/20_209969  2032140141          004123/20         209969   
1  0050153/20_106030  4031560133         0050153/20         106030   

    eviction_address eviction_apartment_number executed_date borough  zipcode  \
0   2541 A GRAND AVE                   ROOM 3B    2022-08-22   BRONX    10468   
1  98-05 67TH AVENUE                       12F    2022-04-14  QUEENS    11375   

          ejectment eviction/legal_possession  latitude  longitude  \
0  Not an Ejectment                Possession   40.8654   -73.9013   
1  Not an Ejectment                Possession   40.7242   -73.8556   

   community_board  council_district  census_tract      bin  \
0           7.0000           14.0000      265.0000  2113173   
1           6.0000           29.0000    71306.0000  4074666   

                   nta  year month_year                      geometry  \
0  Kingsbridge Heights  2022    2022-08  POINT (-73.901317 40.865396)   
1         Forest Hills  2022    2022-04  POINT (-73.855552 40.724241)   

   yearbuilt bldgclass  numfloors  unitsres  \
0  2004.0000        C0     3.0000    3.0000   
1  1960.0000        D3    13.0000  181.0000   

                                ownername    bldgarea building_type  \
0                            MONJU SARKER   3420.0000      post-war   
1  MARSEILLES LEASING LIMITED PARTNERSHIP 177710.0000      post-war   

  building_category  is_condo floor_category                         rent_era  \
0           walk-up     False       low-rise  1994–Present, vacancy decontrol   
1          elevator     False      high-rise          1947–1969, rent-control   

                                 architectural_style  \
0                     2001-present, New Architecture   
1  1951–1980, the International Style, Alternativ...   

                     economic_period residential_units_category  is_llc  \
0  1991–2008, modern economic growth                  3-5 units   False   
1   1946–1975, pst war economic boom                 100+ units   False   

  building_size_category     size_quartile     decade   fips   e_totpop  \
0           medium-small  Q4 (largest 25%)  2000-2009  10468 81397.0000   
1                   mega  Q4 (largest 25%)  1960-1969  11375 75212.0000   

   rpl_theme1  rpl_theme2  rpl_theme3  rpl_theme4  rpl_themes  ep_pov150  \
0      0.9954      0.9407      0.9870      0.9470      0.9874    39.5000   
1      0.4759      0.5698      0.8789      0.8057      0.7322    12.0000   

   ep_unemp  ep_nohsdp  ep_uninsur  ep_age65  ep_age17  ep_disabl  ep_limeng  \
0   11.6000    28.3000      9.2000   11.2000   26.4000    12.2000    26.9000   
1    4.8000     6.1000      3.7000   20.4000   18.0000    10.5000     7.9000   

   ep_noveh  ep_crowd  ep_hburd  ep_afam  ep_hisp  ep_asian  ep_aian  ep_nhpi  \
0   71.8000   19.2000   56.7000  15.6000  78.0000    2.3000   0.0000   0.0000   
1   41.9000    5.8000   25.4000   2.7000  16.4000   28.5000   0.1000   0.0000   

   ep_twomore  ep_otherrace  ep_minrty  ep_white  invalid_zip svi_quartile  \
0      0.5000        0.5000    96.9000    3.1000        False           Q3   
1      4.6000        0.7000    53.0000   47.0000        False     Q1 (Low)   

   air_quality  animal_issues  appliances  building_exterior  doors_windows  \
0       0.0000         0.0000      0.0000             0.0000         3.0000   
1       0.0000         2.0000      0.0000             0.0000         0.0000   

   electrical_issues  elevator_issues  floors_stairs  general_complaints  \
0             0.0000           0.0000         2.0000              0.0000   
1             0.0000           0.0000         0.0000              2.0000   

   graffiti_posting  heat_hot_water  homeless_issues  noise_complaints  \
0            0.0000          1.0000           0.0000            2.0000   
1            0.0000         62.0000           0.0000           34.0000   

   other_issues  pest_issues  plumbing_issues  police_matters  \
0       

In [98]:
evictions_pre_post.columns, \
evictions_covid.columns, \
evictions_pre_post.shape, \
evictions_covid.shape

(Index(['primary_key', 'bbl', 'court_index_number', 'docket_number',
        'eviction_address', 'eviction_apartment_number', 'executed_date',
        'borough', 'zipcode', 'ejectment',
        ...
        'average_year_eviction_per_building_zipcode(z)',
        'evictions_per_zipcode_1kpop(z)',
        'evictions_per_1kunit_zipcode_1kpop(z)',
        'children_impacted_zipcode_per_1kpop(z)',
        'poverty_impacted_zipcode_per_1kpop(z)',
        'disability_impacted_zipcode_per_1kpop(z)',
        'unemployed_impacted_zipcode_per_1kpop(z)',
        'elderly_impacted_zipcode_per_1kpop(z)',
        'bh_impacted_zipcode_per_1kpop(z)',
        'hburd_impacted_zipcode_per_1kpop(z)'],
       dtype='object', length=121),
 Index(['primary_key', 'bbl', 'court_index_number', 'docket_number',
        'eviction_address', 'eviction_apartment_number', 'executed_date',
        'borough', 'zipcode', 'ejectment',
        ...
        'average_year_eviction_per_building_zipcode(z)',
        'evictions_

In [99]:
link = '/content/drive/My Drive/X999/svi_cleaned.csv'

In [100]:
svi_df = pd.read_csv(link)
svi_df.head(2)

fips     location  area_sqmi  e_totpop  m_totpop   e_hu  m_hu   e_hh  \
0  10001  ZCTA5 10001     0.6238     27004      1827  16975   831  14375   
1  10002  ZCTA5 10002     0.8223     76518      2894  39094  1241  36028   

   m_hh  e_pov150  m_pov150  e_unemp  m_unemp  e_hburd  m_hburd  e_nohsdp  \
0   782      5248       797      761      266     3314      531      1930   
1  1326     27908      2853     2833      574    14688     1367     18301   

   m_nohsdp  e_uninsur  m_uninsur  e_age65  m_age65  e_age17  m_age17  \
0       534        831        289     3428      432     2694      643   
1      1376       4074        766    17681     1287    10028     1549   

   e_disabl  m_disabl  e_sngpnt  m_sngpnt  e_limeng  m_limeng  e_minrty  \
0      2310       499       501       215      1381       405     13460   
1      9896      1062      2211       499     18393      1640     56964   

   m_minrty  e_munit  m_munit  e_mobile  m_mobile  e_crowd  m_crowd  e_noveh  \
0      2305    15840      898        15        23      389      135    12285   
1      3226    35725     1677        16        28     2461      449    29828   

   m_noveh  e_groupq  m_groupq  ep_pov150  mp_pov150  ep_unemp  mp_unemp  \
0      840      2213       218    20.3000     2.7000    4.3000    1.5000   
1     1403      2090        39    36.8000     3.5000    7.6000    1.4000   

   ep_hburd  mp_hburd  ep_nohsdp  mp_nohsdp  ep_uninsur  mp_uninsur  ep_age65  \
0   23.1000    3.5000     9.1000     2.4000      3.1000      1.0000   12.7000   
1   40.8000    3.5000    30.0000     2.0000      5.4000      1.0000   23.1000   

   mp_age65  ep_age17  mp_age17  ep_disabl  mp_disabl  ep_sngpnt  mp_sngpnt  \
0    1.6000   10.0000    2.1000     8.6000     1.9000     3.5000     1.5000   
1    1.7000   13.1000    1.8000    13.0000     1.4000     6.1000     1.4000   

   ep_limeng  mp_limeng  ep_minrty  mp_minrty  ep_munit  mp_munit  ep_mobile  \
0     5.3000     1.5000    49.8000     7.8000   93.3000    2.7000     0.1000   
1    24.7000     2.0000    74.4000     3.1000   91.4000    3.2000     0.0000   

   mp_mobile  ep_crowd  mp_crowd  ep_noveh  mp_noveh  ep_groupq  mp_groupq  \
0     0.1000    2.7000    0.9000   85.5000    2.8000     8.2000     0.6000   
1     0.1000    6.8000    1.2000   82.8000    1.8000     2.7000     0.1000   

   epl_pov150  epl_unemp  epl_hburd  epl_nohsdp  epl_uninsur  spl_theme1  \
0      0.6108     0.4574     0.5573      0.5902       0.4436      2.6593   
1      0.9148     0.7946     0.9219      0.9741       0.7207      4.3261   

   rpl_theme1  epl_age65  epl_age17  epl_disabl  epl_sngpnt  epl_limeng  \
0      0.5688     0.1420     0.1161      0.1891      0.4707      0.8777   
1      0.9639     0.7296     0.1831      0.5186      0.7390      0.9944   

   spl_theme2  rpl_theme2  epl_minrty  spl_theme3  rpl_theme3  epl_munit  \
0      1.7956      0.1692      0.8670      0.8670      0.8670     0.9853   
1      3.1647      0.8781      0.9369      0.9369      0.9369     0.9790   

   epl_mobile  epl_crowd  epl_noveh  epl_groupq  spl_theme4  rpl_theme4  \
0      0.2710     0.7402     0.9949      0.9104      3.9018      0.9806   
1      0.0000     0.9105     0.9915      0.7730      3.6540      0.9254   

   spl_themes  rpl_themes  f_pov150  f_unemp  f_hburd  f_nohsdp  f_uninsur  \
0      9.2237      0.7414         0        0        0         0          0   
1     12.0817      0.9656         1        0        1         1          0   

   f_theme1  f_age65  f_age17  f_disabl  f_sngpnt  f_limeng  f_theme2  \
0         0        0        0         0         0         0         0   
1         3        0        0         0         0         1         1   

   f_minrty  f_theme3  f_munit  f_mobile  f_crowd  f_noveh  f_groupq  \
0         0         0        1         0        0        1         1   
1         1         1        1         0        1        1         0   

   f_theme4  f_total  e_daypop  e_noint  m_noint  e_afam  m_afam  e_hisp  \
0         3

In [101]:
svi_df.shape

(204, 153)

In [102]:
# list(svi_df.columns)

In [103]:
svi_df.ep_nhpi.unique()

array([ 0.00e+00,  1.00e-01,  3.00e-01,  2.00e-01, -9.99e+02,  8.00e-01,
        1.20e+00,  5.00e-01,  4.00e-01])

# **Step 2: SVI items**

q quick double check

In [104]:
link = "/content/drive/My Drive/X999/NewYork_ZCTA.csv"

In [105]:
svi_raw = pd.read_csv(link)
svi_raw.head(2)

ST     STATE ST_ABBR   FIPS     LOCATION  AREA_SQMI  E_TOTPOP  M_TOTPOP  \
0  36  New York      NY   6390  ZCTA5 06390     4.0467        53        39   
1  36  New York      NY  10001  ZCTA5 10001     0.6238     27004      1827   

    E_HU  M_HU   E_HH  M_HH  E_POV150  M_POV150  E_UNEMP  M_UNEMP  E_HBURD  \
0    253    49     19    19        17        16        0       13        9   
1  16975   831  14375   782      5248       797      761      266     3314   

   M_HBURD  E_NOHSDP  M_NOHSDP  E_UNINSUR  M_UNINSUR  E_AGE65  M_AGE65  \
0       26         0        13         27         34        0       13   
1      531      1930       534        831        289     3428      432   

   E_AGE17  M_AGE17  E_DISABL  M_DISABL  E_SNGPNT  M_SNGPNT  E_LIMENG  \
0        6       11        31        33         0        18         9   
1     2694      643      2310       499       501       215      1381   

   M_LIMENG  E_MINRTY  M_MINRTY  E_MUNIT  M_MUNIT  E_MOBILE  M_MOBILE  \
0        53        20        51        0       18         4         5   
1       405     13460      2305    15840      898        15        23   

   E_CROWD  M_CROWD  E_NOVEH  M_NOVEH  E_GROUPQ  M_GROUPQ  EP_POV150  \
0        0       18        0       13        17        16    32.1000   
1      389      135    12285      840      2213       218    20.3000   

   MP_POV150  EP_UNEMP  MP_UNEMP  EP_HBURD  MP_HBURD  EP_NOHSDP  MP_NOHSDP  \
0    18.8000    0.0000   52.7000   47.4000  100.0000     0.0000    51.4000   
1     2.7000    4.3000    1.5000   23.1000    3.5000     9.1000     2.4000   

   EP_UNINSUR  MP_UNINSUR  EP_AGE65  MP_AGE65  EP_AGE17  MP_AGE17  EP_DISABL  \
0     50.9000     45.7000    0.0000   45.2000   11.3000   19.0000    58.5000   
1      3.1000      1.0000   12.7000    1.6000   10.0000    2.1000     8.6000   

   MP_DISABL  EP_SNGPNT  MP_SNGPNT  EP_LIMENG  MP_LIMENG  EP_MINRTY  \
0    41.4000     0.0000    94.7000    17.0000    99.2000    37.7000   
1     1.9000     3.5000     1.5000     5.3000     1.5000    49.8000   

   MP_MINRTY  EP_MUNIT  MP_MUNIT  EP_MOBILE  MP_MOBILE  EP_CROWD  MP_CROWD  \
0    92.1000    0.0000    7.1000     1.6000     2.0000    0.0000   94.7000   
1     7.8000   93.3000    2.7000     0.1000     0.1000    2.7000    0.9000   

   EP_NOVEH  MP_NOVEH  EP_GROUPQ  MP_GROUPQ  EPL_POV150  EPL_UNEMP  EPL_HBURD  \
0    0.0000   75.5000    32.1000    18.8000      0.8790     0.0000     0.9635   
1   85.5000    2.8000     8.2000     0.6000      0.6108     0.4574     0.5573   

   EPL_NOHSDP  EPL_UNINSUR  SPL_THEME1  RPL_THEME1  EPL_AGE65  EPL_AGE17  \
0      0.0000       0.9960      2.8385      0.6342     0.0000     0.1408   
1      0.5902       0.4436      2.6593      0.5688     0.1420     0.1161   

   EPL_DISABL  EPL_SNGPNT  EPL_LIMENG  SPL_THEME2  RPL_THEME2  EPL_MINRTY  \
0      0.9944      0.0000      0.9775      2.1127      0.3009      0.8062   
1      0.1891      0.4707      0.8777      1.7956      0.1692      0.8670   

   SPL_THEME3  RPL_THEME3  EPL_MUNIT  EPL_MOBILE  EPL_CROWD  EPL_NOVEH  \
0      0.8062      0.8062     0.0000      0.4654     0.0000     0.0000   
1      0.8670      0.8670     0.9853      0.2710     0.7402     0.9949   

   EPL_GROUPQ  SPL_THEME4  RPL_THEME4  SPL_THEMES  RPL_THEMES  F_POV150  \
0      0.9735      1.4389      0.2205      7.1963      0.4192         0   
1      0.9104      3.9018      0.9806      9.2237      0.7414         0   

   F_UNEMP  F_HBURD  F_NOHSDP  F_UNINSUR  F_THEME1  F_AGE65  F_AGE17  \
0        0        1         0          1         2        0        0   
1        0        0         0          0         0        0        0   

   F_DISABL  F_SNGPNT  F_LIMENG  F_THEME2  F_MINRTY  F_THEME3  F_MUNIT  \
0         1         0         1         2         0         0        0   
1         0         0         0         0         0         0        1   

   F_MOBILE  F_CROWD  F_NOVEH  F_GROUPQ  F_THEME4  F_TOTAL  E_DAYPOP  E_NOINT  \
0         0        0        0         1         1    

In [106]:
def is_nyc_zipcode(zipcode):
    zip_int = int(zipcode) if isinstance(zipcode, str) else zipcode

    # Manhattan: 10001-10282
    if 10001 <= zip_int <= 10282:
        return True
    # addition Manhattan: 10300-10499
    if 10300 <= zip_int <= 10499:
        return True
    # Bronx: 10451-10475
    if 10451 <= zip_int <= 10475:
        return True
    # Brooklyn: 11201-11256
    if 11201 <= zip_int <= 11256:
        return True
    # Queens: 11351-11436, 11101-11109
    if (11351 <= zip_int <= 11436) or (11101 <= zip_int <= 11109):
        return True
    # Staten Island: 10301-10314
    if 10301 <= zip_int <= 10314:
        return True
    # additional Queens ZIPs
    if zip_int in [11004, 11005, 11411, 11412, 11413, 11418, 11419, 11420, 11421, 11422, 11423, 11426, 11427, 11428, 11429]:
        return True
    return False

In [107]:
nyc_df = svi_raw[svi_raw['FIPS'].apply(is_nyc_zipcode)]

In [108]:
nyc_df.shape

(204, 156)

In [109]:
nyc_df.EP_NHPI.unique()
# interesting, so there was an error in the source data

array([ 0.00e+00,  1.00e-01,  3.00e-01,  2.00e-01, -9.99e+02,  8.00e-01,
        1.20e+00,  5.00e-01,  4.00e-01])

In [110]:
svi_raw.EP_NHPI.unique()

array([ 1.51e+01,  0.00e+00,  1.00e-01,  3.00e-01,  2.00e-01, -9.99e+02,
        8.00e-01,  1.20e+00,  5.00e-01,  4.00e-01,  1.10e+00,  7.00e-01,
        1.50e+00,  1.80e+00,  9.00e-01,  6.00e-01,  1.40e+00,  2.10e+00,
        2.30e+00,  1.00e+00])

In [111]:
-9.99e+02, 0.00e+00, 9.00e-01, 2.00e-01

(-999.0, 0.0, 0.9, 0.2)

# **Step 3: All boroughs and their eviction rates**

In [112]:
evictions_pre_post_mean = evictions_pre_post[['ep_afam', 'ep_asian', 'ep_hisp', 'ep_nhpi', 'ep_white', 'ep_twomore', 'ep_otherrace']].mean()
evictions_pre_post_mean
# may need to merge ep_twomore and ep_otherrance toegther

ep_afam        29.2498
ep_asian        8.9193
ep_hisp        38.7465
ep_nhpi         0.0265
ep_white       19.0073
ep_twomore      2.8176
ep_otherrace    1.0390
dtype: float64

In [113]:
evictions_pre_post_mean = evictions_pre_post_mean.reset_index()

In [114]:
evictions_pre_post_mean.rename(columns = {'index':'race_svi', 0: "racial percezipcodege"}, inplace=True)

In [115]:
evictions_pre_post_mean

race_svi  racial percezipcodege
0       ep_afam                29.2498
1      ep_asian                 8.9193
2       ep_hisp                38.7465
3       ep_nhpi                 0.0265
4      ep_white                19.0073
5    ep_twomore                 2.8176
6  ep_otherrace                 1.0390

In [116]:
# type(evictions_pre_post_mean)
# so this is correct

In [117]:
# evictions_pre_post.columns

In [118]:
evictions_pre_post.columns[-30:]

Index(['average_year_eviction_count_per_unit(building)',
       'average_year_eviction_count(building)',
       'average_year_eviction_count_per_nta(nta)',
       'evictions_per_nta_1kpop(nta)', 'buildings_affected_per_nta',
       'unit_per_nta', 'evictions_per_1kunit_nta_1kpop(nta)',
       'children_impacted_nta_per_1kpop', 'poverty_impacted_nta_per_1kpop',
       'disability_impacted_nta_per_1kpop',
       'unemployed_impacted_nta_per_1kpop', 'elderly_impacted_nta_per_1kpop',
       'bh_impacted_nta_per_1kpop', 'hburd_impacted_nta_per_1kpop',
       'average_year_eviction_per_building_nta(nta)',
       'average_5year_eviction_count_per_unit(building)', 'bh_pct',
       'unit_count_zipcode', 'building_count_zipcode',
       'average_year_eviction_count_per_zipcode(z)',
       'average_year_eviction_per_building_zipcode(z)',
       'evictions_per_zipcode_1kpop(z)',
       'evictions_per_1kunit_zipcode_1kpop(z)',
       'children_impacted_zipcode_per_1kpop(z)',
       'poverty_impacte

In [121]:
evictions_pre_post['average_year_eviction_per_building_zipcode(z)'].mean()
# per building, per year, that's why it is similar to the borough one too

np.float64(0.46716589822318577)

In [122]:
evictions_pre_post['average_year_eviction_count_per_zipcode(z)'].mean()

np.float64(909.649898335718)

### **Step 3.2. All zipcodes racial composite**

In [137]:
race_columns = ['ep_afam', 'ep_asian', 'ep_hisp', 'ep_nhpi', 'ep_white', 'ep_twomore', 'ep_otherrace']
racial_avg_all_zt = evictions_pre_post.groupby('zipcode')[['average_year_eviction_count_per_zipcode(z)', 'ep_afam', 'ep_asian', 'ep_hisp', 'ep_nhpi', 'ep_white', 'ep_twomore', 'ep_otherrace', 'rpl_themes']].mean()
racial_avg_all_zt.sort_values('average_year_eviction_count_per_zipcode(z)', ascending=False, inplace=True)
racial_avg_all_zt.reset_index(inplace=True)
racial_avg_all_zt

zipcode  average_year_eviction_count_per_zipcode(z)  ep_afam  ep_asian  \
0      10458                                   2221.0000  16.1000    3.3000   
1      10467                                   2192.0000  28.6000    6.2000   
2      10453                                   1994.0000  24.6000    1.6000   
3      10456                                   1880.0000  38.2000    0.7000   
4      10457                                   1765.0000  30.9000    0.4000   
5      10468                                   1719.0000  15.6000    2.3000   
6      11226                                   1639.0000  63.2000    3.2000   
7      10452                                   1592.0000  28.1000    1.0000   
8      10460                                   1557.0000  26.5000    1.4000   
9      11212                                   1487.0000  70.1000    0.8000   
10     11207                                   1320.0000  55.9000    1.5000   
11     11233                                   1196.0000  66.7000    1.3000   
12     11208                                   1129.0000  45.8000    8.1000   
13     10466                                   1119.0000  59.8000    2.6000   
14     10472                                   1070.0000  23.1000    8.2000   
15     10459                                   1041.0000  28.9000    0.2000   
16     11213                                   1011.0000  56.6000    1.6000   
17     11203                                    952.0000  80.3000    1.8000   
18     10451                                    932.0000  37.2000    1.4000   
19     10462                                    922.0000  22.2000   16.6000   
20     11221                                    897.0000  39.7000    5.1000   
21     11210                                    843.0000  51.6000    5.8000   
22     11368                                    810.0000   7.4000   11.2000   
23     10463                                    746.0000  12.1000    3.7000   
24     10029                                    724.0000  25.0000   10.8000   
25     10032                                    689.0000  12.8000    4.5000   
26     11225                                    682.0000  53.7000    3.3000   
27     10031                                    678.0000  23.6000    4.0000   
28     10455                                    664.0000  21.1000    1.1000   
29     10033                                    627.0000   3.6000    3.3000   
30     11236                                    612.0000  79.6000    2.5000   
31     10469                                    602.0000  52.2000    5.5000   
32     10034                                    579.0000   5.7000    3.3000   
33     11216                                    559.0000  44.1000    4.6000   
34     10454                                    546.0000  24.6000    0.2000   
35     10030                                    546.0000  53.8000    3.6000   
36     10473                                    544.0000  30.9000    2.1000   
37     11206                                    543.0000  19.0000    6.4000   
38     10027                                    532.0000  33.9000    8.9000   
39     10040                                    513.0000   2.9000    1.8000   
40     11230                                    513.0000   8.2000   15.9000   
41     10039                                    506.0000  54.7000    1.9000   
42     10037                                    506.0000  63.1000    3.8000   
43     11434                                    494.0000  78.0000    3.0000   
44     10035                                    470.0000  36.7000    3.4000   
45     11235                                    470.0000   2.4000   14.8000   
46     11432                                    468.0000  16.8000   42.8000   
47     10461                                    448.0000   7.7000   12.8000   
48     10301                                    445.0000  19.9000    7.6000   
49     10304                                    442.000

In [138]:
racial_avg_all_zt.T

0          1          2    \
zipcode                                    10458.0000 10467.0000 10453.0000   
average_year_eviction_count_per_zipcode(z)  2221.0000  2192.0000  1994.0000   
ep_afam                                       16.1000    28.6000    24.6000   
ep_asian                                       3.3000     6.2000     1.6000   
ep_hisp                                       71.3000    52.9000    69.2000   
ep_nhpi                                        0.0000     0.0000     0.0000   
ep_white                                       7.0000     8.7000     2.0000   
ep_twomore                                     1.7000     2.3000     1.9000   
ep_otherrace                                   0.4000     0.7000     0.7000   
rpl_themes                                     0.9891     0.9925     0.9983   

                                                  3          4          5    \
zipcode                                    10456.0000 10457.0000 10468.0000   
average_year_eviction_count_per_zipcode(z)  1880.0000  1765.0000  1719.0000   
ep_afam                                       38.2000    30.9000    15.6000   
ep_asian                                       0.7000     0.4000     2.3000   
ep_hisp                                       56.3000    65.3000    78.0000   
ep_nhpi                                        0.0000     0.0000     0.0000   
ep_white                                       2.4000     1.7000     3.1000   
ep_twomore                                     1.8000     1.1000     0.5000   
ep_otherrace                                   0.4000     0.4000     0.5000   
rpl_themes                                     0.9994     0.9989     0.9874   

                                                  6          7          8    \
zipcode                                    11226.0000 10452.0000 10460.0000   
average_year_eviction_count_per_zipcode(z)  1639.0000  1592.0000  1557.0000   
ep_afam                                       63.2000    28.1000    26.5000   
ep_asian                                       3.2000     1.0000     1.4000   
ep_hisp                                       14.9000    65.4000    65.0000   
ep_nhpi                                        0.0000     0.0000     0.0000   
ep_white                                      13.7000     2.7000     3.8000   
ep_twomore                                     4.1000     1.4000     1.9000   
ep_otherrace                                   0.7000     1.1000     1.0000   
rpl_themes                                     0.9220     0.9931     0.9954   

                                                  9          10         11   \
zipcode                                    11212.0000 11207.0000 11233.0000   
average_year_eviction_count_per_zipcode(z)  1487.0000  1320.0000  1196.0000   
ep_afam                                       70.1000    55.9000    66.7000   
ep_asian                                       0.8000     1.5000     1.3000   
ep_hisp                                       19.5000    32.8000    15.7000   
ep_nhpi                                        0.0000     0.0000     0.0000   
ep_white                                       3.5000     5.3000    11.9000   
ep_twomore                                     5.4000     2.9000     3.6000   
ep_otherrace                                   0.6000     1.6000     0.2000   
rpl_themes                                     0.9948     0.9839     0.9438   

                                                  12         13         14   \
zipcode                                    11208.0000 10466.0000 10472.0000   
average_year_eviction_count_per_zipcode(z)  1129.0000  1119.0000  1070.0000   
ep_afam                                       45.8000    59.8000    23.1000   
ep_asian                                       8.1000     2.6000     8.2000   
ep_hisp                                       39.0000    29.4000    61.7000   
ep_nhpi                                        0.2000     0.0000     0.1000   
ep_white         

In [142]:
race_columns = ['ep_afam', 'ep_asian', 'ep_hisp', 'ep_nhpi', 'ep_white', 'ep_twomore', 'ep_otherrace']
racial_avg_all_zp = evictions_pre_post.groupby('zipcode')[['evictions_per_zipcode_1kpop(z)', 'ep_afam', 'ep_asian', 'ep_hisp', 'ep_nhpi', 'ep_white', 'ep_twomore', 'ep_otherrace', 'rpl_themes']].mean()
racial_avg_all_zp.sort_values('evictions_per_zipcode_1kpop(z)', ascending=False, inplace=True)
racial_avg_all_zp.reset_index(inplace=True)
racial_avg_all_zp

# 'evictions_per_zipcode_1kpop(z)',
# 'evictions_per_1kunit_zipcode_1kpop(z)'

zipcode  evictions_per_zipcode_1kpop(z)  ep_afam  ep_asian  ep_hisp  \
0      10458                         26.8633  16.1000    3.3000  71.3000   
1      10037                         26.4395  63.1000    3.8000  19.8000   
2      10474                         26.3298  26.4000    0.2000  70.2000   
3      10460                         26.0142  26.5000    1.4000  65.0000   
4      10453                         24.8056  24.6000    1.6000  69.2000   
5      10467                         22.2058  28.6000    6.2000  52.9000   
6      10457                         22.1131  30.9000    0.4000  65.3000   
7      10456                         21.2250  38.2000    0.7000  56.3000   
8      10468                         21.1187  15.6000    2.3000  78.0000   
9      10452                         20.2483  28.1000    1.0000  65.4000   
10     10459                         20.0331  28.9000    0.2000  67.3000   
11     10451                         18.1637  37.2000    1.4000  55.0000   
12     11212                         17.7011  70.1000    0.8000  19.5000   
13     10030                         17.4619  53.8000    3.6000  29.7000   
14     10470                         17.3489  31.3000    2.0000  22.8000   
15     10039                         16.9304  54.7000    1.9000  32.7000   
16     10472                         16.3902  23.1000    8.2000  61.7000   
17     11226                         16.2192  63.2000    3.2000  14.9000   
18     10466                         15.4830  59.8000    2.6000  29.4000   
19     10455                         14.9617  21.1000    1.1000  74.1000   
20     11213                         14.6244  56.6000    1.6000  14.3000   
21     11233                         14.3880  66.7000    1.3000  15.7000   
22     10034                         13.9283   5.7000    3.3000  69.0000   
23     11207                         13.6362  55.9000    1.5000  32.8000   
24     10454                         13.2934  24.6000    0.2000  70.3000   
25     11210                         13.2911  51.6000    5.8000   7.5000   
26     11239                         13.1485  61.7000    2.6000  15.9000   
27     10035                         12.1777  36.7000    3.4000  43.0000   
28     11203                         12.1265  80.3000    1.8000   6.9000   
29     10462                         11.9384  22.2000   16.6000  47.2000   
30     10040                         11.7518   2.9000    1.8000  71.8000   
31     11225                         11.6629  53.7000    3.3000  10.8000   
32     10032                         11.3922  12.8000    4.5000  65.7000   
33     10031                         11.1067  23.6000    4.0000  50.9000   
34     10301                         11.0337  19.9000    7.6000  26.3000   
35     10033                         10.6794   3.6000    3.3000  67.2000   
36     10463                         10.6123  12.1000    3.7000  52.3000   
37     11208                         10.4363  45.8000    8.1000  39.0000   
38     10026                         10.0091  47.6000    6.3000  19.8000   
39     10304                          9.8520  26.0000   12.3000  21.7000   
40     11221                          9.8316  39.7000    5.1000  31.0000   
41     10029                          9.5749  25.0000   10.8000  45.7000   
42     11224                          9.1871  20.1000    9.5000  19.5000   
43     10473                          9.0535  30.9000    2.1000  62.2000   
44     11216                          8.9622  44.1000    4.6000  13.6000   
45     10461                          8.8071   7.7000   12.8000  49.7000   
46     11415                          8.7620   6.7000   22.3000  22.9000   
47     10469                          8.3772  52.2000    5.5000  29.5000   
48     11433                          8.3623  55.3000   15.4000  13.9000   
49     10303                          8.2339  30.0000    9.8000  39.0000   
50     11423                          8.1548  33.4000   30.6000  14.7000   
51     10019                          8.0905   4.5000

In [143]:
racial_avg_all_zp.T

0          1          2          3    \
zipcode                        10458.0000 10037.0000 10474.0000 10460.0000   
evictions_per_zipcode_1kpop(z)    26.8633    26.4395    26.3298    26.0142   
ep_afam                           16.1000    63.1000    26.4000    26.5000   
ep_asian                           3.3000     3.8000     0.2000     1.4000   
ep_hisp                           71.3000    19.8000    70.2000    65.0000   
ep_nhpi                            0.0000     0.0000     0.0000     0.0000   
ep_white                           7.0000     9.6000     2.2000     3.8000   
ep_twomore                         1.7000     2.9000     0.3000     1.9000   
ep_otherrace                       0.4000     0.8000     0.2000     1.0000   
rpl_themes                         0.9891     0.9639     0.9960     0.9954   

                                      4          5          6          7    \
zipcode                        10453.0000 10467.0000 10457.0000 10456.0000   
evictions_per_zipcode_1kpop(z)    24.8056    22.2058    22.1131    21.2250   
ep_afam                           24.6000    28.6000    30.9000    38.2000   
ep_asian                           1.6000     6.2000     0.4000     0.7000   
ep_hisp                           69.2000    52.9000    65.3000    56.3000   
ep_nhpi                            0.0000     0.0000     0.0000     0.0000   
ep_white                           2.0000     8.7000     1.7000     2.4000   
ep_twomore                         1.9000     2.3000     1.1000     1.8000   
ep_otherrace                       0.7000     0.7000     0.4000     0.4000   
rpl_themes                         0.9983     0.9925     0.9989     0.9994   

                                      8          9          10         11   \
zipcode                        10468.0000 10452.0000 10459.0000 10451.0000   
evictions_per_zipcode_1kpop(z)    21.1187    20.2483    20.0331    18.1637   
ep_afam                           15.6000    28.1000    28.9000    37.2000   
ep_asian                           2.3000     1.0000     0.2000     1.4000   
ep_hisp                           78.0000    65.4000    67.3000    55.0000   
ep_nhpi                            0.0000     0.0000     0.0000     0.0000   
ep_white                           3.1000     2.7000     1.9000     4.5000   
ep_twomore                         0.5000     1.4000     1.3000     1.3000   
ep_otherrace                       0.5000     1.1000     0.1000     0.3000   
rpl_themes                         0.9874     0.9931     0.9943     0.9897   

                                      12         13         14         15   \
zipcode                        11212.0000 10030.0000 10470.0000 10039.0000   
evictions_per_zipcode_1kpop(z)    17.7011    17.4619    17.3489    16.9304   
ep_afam                           70.1000    53.8000    31.3000    54.7000   
ep_asian                           0.8000     3.6000     2.0000     1.9000   
ep_hisp                           19.5000    29.7000    22.8000    32.7000   
ep_nhpi                            0.0000     0.0000     0.0000     0.0000   
ep_white                           3.5000     8.9000    38.8000     6.7000   
ep_twomore                         5.4000     3.0000     3.2000     2.8000   
ep_otherrace                       0.6000     1.0000     1.7000     1.1000   
rpl_themes                         0.9948     0.9828     0.9398     0.9748   

                                      16         17         18         19   \
zipcode                        10472.0000 11226.0000 10466.0000 10455.0000   
evictions_per_zipcode_1kpop(z)    16.3902    16.2192    15.4830    14.9617   
ep_afam                           23.1000    63.2000    59.8000    21.1000   
ep_asian                           8.2000     3.2000     2.6000     1.1000   
ep_hisp                           61.7000    14.9000    29.4000    74.1000   
ep_nhpi                            0.1000     0.0000     0.0000     0.0000   
ep_white                           2.1000    13.7000    

In [146]:
race_columns = ['ep_afam', 'ep_asian', 'ep_hisp', 'ep_nhpi', 'ep_white', 'ep_twomore', 'ep_otherrace']
racial_avg_all_zup = evictions_pre_post.groupby('zipcode')[['evictions_per_1kunit_zipcode_1kpop(z)', 'ep_afam', 'ep_asian', 'ep_hisp', 'ep_nhpi', 'ep_white', 'ep_twomore', 'ep_otherrace', 'rpl_themes']].mean()
racial_avg_all_zup.sort_values('evictions_per_1kunit_zipcode_1kpop(z)', ascending=False, inplace=True)
racial_avg_all_zup.reset_index(inplace=True)
racial_avg_all_zup

# 'evictions_per_zipcode_1kpop(z)',
# 'evictions_per_1kunit_zipcode_1kpop(z)'

zipcode  evictions_per_1kunit_zipcode_1kpop(z)  ep_afam  ep_asian  \
0      10464                                 0.0163   0.7000    2.4000   
1      11411                                 0.0122  86.8000    1.5000   
2      11366                                 0.0120   4.9000   42.3000   
3      10307                                 0.0091   0.1000    2.1000   
4      11436                                 0.0090  61.3000    6.4000   
5      11428                                 0.0073  20.0000   31.8000   
6      10302                                 0.0068  21.7000    5.6000   
7      11429                                 0.0064  65.4000    7.4000   
8      11004                                 0.0064   5.3000   51.4000   
9      11412                                 0.0060  79.9000    3.1000   
10     10470                                 0.0058  31.3000    2.0000   
11     11426                                 0.0056   5.7000   46.8000   
12     11356                                 0.0055   0.5000   32.8000   
13     11422                                 0.0052  79.2000    2.4000   
14     11433                                 0.0049  55.3000   15.4000   
15     11417                                 0.0048   7.1000   28.2000   
16     10474                                 0.0047  26.4000    0.2000   
17     11369                                 0.0047  16.3000   14.9000   
18     11362                                 0.0045   2.5000   48.9000   
19     10303                                 0.0045  30.0000    9.8000   
20     11416                                 0.0044   3.5000   34.0000   
21     11413                                 0.0043  85.7000    1.6000   
22     11420                                 0.0040  21.3000   33.4000   
23     10309                                 0.0038   0.5000    4.1000   
24     10308                                 0.0037   0.7000    6.9000   
25     10310                                 0.0037  20.0000    5.3000   
26     11414                                 0.0036   3.6000    5.6000   
27     11419                                 0.0033  13.9000   34.0000   
28     11370                                 0.0033  10.6000   24.5000   
29     11361                                 0.0030   3.7000   40.3000   
30     11427                                 0.0029  21.5000   34.5000   
31     11423                                 0.0028  33.4000   30.6000   
32     11363                                 0.0028   0.3000   30.9000   
33     11421                                 0.0026   4.1000   17.9000   
34     11378                                 0.0026   0.9000   13.5000   
35     10312                                 0.0026   0.8000    8.7000   
36     11418                                 0.0024   7.5000   23.4000   
37     10305                                 0.0024   4.4000   20.1000   
38     11357                                 0.0024   0.3000   26.0000   
39     10465                                 0.0024  10.1000    4.2000   
40     11379                                 0.0022   0.6000   12.3000   
41     10301                                 0.0019  19.9000    7.6000   
42     10162                                 0.0019   0.6000    2.9000   
43     11228                                 0.0018   0.9000   35.1000   
44     11232                                 0.0017   6.0000   15.4000   
45     11358                                 0.0016   0.7000   48.0000   
46     11434                                 0.0016  78.0000    3.0000   
47     11415                                 0.0016   6.7000   22.3000   
48     11236                                 0.0016  79.6000    2.5000   
49     10304                                 0.0016  26.0000   12.3000   
50     11364                                 0.0015   1.6000   48.6000   
51     10018                                 0.0015   6.2000   27.6000   
52     11234                                 0.0014  42.6000    8.1000   
53     10

In [145]:
racial_avg_all_zup.T

0          1          2          3    \
zipcode                        10458.0000 10037.0000 10474.0000 10460.0000   
evictions_per_zipcode_1kpop(z)    26.8633    26.4395    26.3298    26.0142   
ep_afam                           16.1000    63.1000    26.4000    26.5000   
ep_asian                           3.3000     3.8000     0.2000     1.4000   
ep_hisp                           71.3000    19.8000    70.2000    65.0000   
ep_nhpi                            0.0000     0.0000     0.0000     0.0000   
ep_white                           7.0000     9.6000     2.2000     3.8000   
ep_twomore                         1.7000     2.9000     0.3000     1.9000   
ep_otherrace                       0.4000     0.8000     0.2000     1.0000   
rpl_themes                         0.9891     0.9639     0.9960     0.9954   

                                      4          5          6          7    \
zipcode                        10453.0000 10467.0000 10457.0000 10456.0000   
evictions_per_zipcode_1kpop(z)    24.8056    22.2058    22.1131    21.2250   
ep_afam                           24.6000    28.6000    30.9000    38.2000   
ep_asian                           1.6000     6.2000     0.4000     0.7000   
ep_hisp                           69.2000    52.9000    65.3000    56.3000   
ep_nhpi                            0.0000     0.0000     0.0000     0.0000   
ep_white                           2.0000     8.7000     1.7000     2.4000   
ep_twomore                         1.9000     2.3000     1.1000     1.8000   
ep_otherrace                       0.7000     0.7000     0.4000     0.4000   
rpl_themes                         0.9983     0.9925     0.9989     0.9994   

                                      8          9          10         11   \
zipcode                        10468.0000 10452.0000 10459.0000 10451.0000   
evictions_per_zipcode_1kpop(z)    21.1187    20.2483    20.0331    18.1637   
ep_afam                           15.6000    28.1000    28.9000    37.2000   
ep_asian                           2.3000     1.0000     0.2000     1.4000   
ep_hisp                           78.0000    65.4000    67.3000    55.0000   
ep_nhpi                            0.0000     0.0000     0.0000     0.0000   
ep_white                           3.1000     2.7000     1.9000     4.5000   
ep_twomore                         0.5000     1.4000     1.3000     1.3000   
ep_otherrace                       0.5000     1.1000     0.1000     0.3000   
rpl_themes                         0.9874     0.9931     0.9943     0.9897   

                                      12         13         14         15   \
zipcode                        11212.0000 10030.0000 10470.0000 10039.0000   
evictions_per_zipcode_1kpop(z)    17.7011    17.4619    17.3489    16.9304   
ep_afam                           70.1000    53.8000    31.3000    54.7000   
ep_asian                           0.8000     3.6000     2.0000     1.9000   
ep_hisp                           19.5000    29.7000    22.8000    32.7000   
ep_nhpi                            0.0000     0.0000     0.0000     0.0000   
ep_white                           3.5000     8.9000    38.8000     6.7000   
ep_twomore                         5.4000     3.0000     3.2000     2.8000   
ep_otherrace                       0.6000     1.0000     1.7000     1.1000   
rpl_themes                         0.9948     0.9828     0.9398     0.9748   

                                      16         17         18         19   \
zipcode                        10472.0000 11226.0000 10466.0000 10455.0000   
evictions_per_zipcode_1kpop(z)    16.3902    16.2192    15.4830    14.9617   
ep_afam                           23.1000    63.2000    59.8000    21.1000   
ep_asian                           8.2000     3.2000     2.6000     1.1000   
ep_hisp                           61.7000    14.9000    29.4000    74.1000   
ep_nhpi                            0.1000     0.0000     0.0000     0.0000   
ep_white                           2.1000    13.7000    

## **This is for the race composites and zipcodes bar char use**

# **Step 4 We want a dataframe that has zipcodes as columns and average_year_eviction_zipcode_count as rows and contents**

In [148]:
evi_svi_df = evictions_pre_post.groupby('zipcode')[['average_year_eviction_count_per_zipcode(z)','rpl_themes']].mean()
evi_svi_df.sort_values('average_year_eviction_count_per_zipcode(z)', ascending=False, inplace=True)
evi_svi_df

average_year_eviction_count_per_zipcode(z)  rpl_themes
zipcode                                                        
10458                                     2221.0000      0.9891
10467                                     2192.0000      0.9925
10453                                     1994.0000      0.9983
10456                                     1880.0000      0.9994
10457                                     1765.0000      0.9989
10468                                     1719.0000      0.9874
11226                                     1639.0000      0.9220
10452                                     1592.0000      0.9931
10460                                     1557.0000      0.9954
11212                                     1487.0000      0.9948
11207                                     1320.0000      0.9839
11233                                     1196.0000      0.9438
11208                                     1129.0000      0.9581
10466                                     1119.0000      0.9851
10472                                     1070.0000      0.9937
10459                                     1041.0000      0.9943
11213                                     1011.0000      0.9805
11203                                      952.0000      0.9386
10451                                      932.0000      0.9897
10462                                      922.0000      0.9708
11221                                      897.0000      0.9352
11210                                      843.0000      0.9174
11368                                      810.0000      0.9633
10463                                      746.0000      0.9862
10029                                      724.0000      0.9788
10032                                      689.0000      0.9771
11225                                      682.0000      0.8538
10031                                      678.0000      0.9323
10455                                      664.0000      0.9971
10033                                      627.0000      0.9650
11236                                      612.0000      0.9117
10469                                      602.0000      0.9507
10034                                      579.0000      0.9541
11216                                      559.0000      0.8297
10454                                      546.0000      1.0000
10030                                      546.0000      0.9828
10473                                      544.0000      0.9868
11206                                      543.0000      0.9679
10027                                      532.0000      0.9312
10040                                      513.0000      0.9857
11230                                      513.0000      0.9702
10039                                      506.0000      0.9748
10037                                      506.0000      0.9639
11434                                      494.0000      0.9564
10035                                      470.0000      0.9914
11235                                      470.0000      0.9524
11432                                      468.0000      0.9713
10461                                      448.0000      0.9690
10301                                      445.0000      0.9329
10304                                      442.0000      0.9467
11224                                      440.0000      0.9920
11372                                      413.0000      0.9255
11435                                      412.0000      0.9518
11385                                      395.0000      0.8819
11214                                      387.0000      0.9685
10026                                      383.0000      0.9123
11373                                      370.0000      0.9432
11209                                      369.0000      0.8865
10025                                      349.0000      0.8607
11377                                      347.0000      0.9019
10019                                    

In [149]:
evi_svi_df.reset_index(inplace=True)
evi_svi_df

zipcode  average_year_eviction_count_per_zipcode(z)  rpl_themes
0      10458                                   2221.0000      0.9891
1      10467                                   2192.0000      0.9925
2      10453                                   1994.0000      0.9983
3      10456                                   1880.0000      0.9994
4      10457                                   1765.0000      0.9989
5      10468                                   1719.0000      0.9874
6      11226                                   1639.0000      0.9220
7      10452                                   1592.0000      0.9931
8      10460                                   1557.0000      0.9954
9      11212                                   1487.0000      0.9948
10     11207                                   1320.0000      0.9839
11     11233                                   1196.0000      0.9438
12     11208                                   1129.0000      0.9581
13     10466                                   1119.0000      0.9851
14     10472                                   1070.0000      0.9937
15     10459                                   1041.0000      0.9943
16     11213                                   1011.0000      0.9805
17     11203                                    952.0000      0.9386
18     10451                                    932.0000      0.9897
19     10462                                    922.0000      0.9708
20     11221                                    897.0000      0.9352
21     11210                                    843.0000      0.9174
22     11368                                    810.0000      0.9633
23     10463                                    746.0000      0.9862
24     10029                                    724.0000      0.9788
25     10032                                    689.0000      0.9771
26     11225                                    682.0000      0.8538
27     10031                                    678.0000      0.9323
28     10455                                    664.0000      0.9971
29     10033                                    627.0000      0.9650
30     11236                                    612.0000      0.9117
31     10469                                    602.0000      0.9507
32     10034                                    579.0000      0.9541
33     11216                                    559.0000      0.8297
34     10454                                    546.0000      1.0000
35     10030                                    546.0000      0.9828
36     10473                                    544.0000      0.9868
37     11206                                    543.0000      0.9679
38     10027                                    532.0000      0.9312
39     10040                                    513.0000      0.9857
40     11230                                    513.0000      0.9702
41     10039                                    506.0000      0.9748
42     10037                                    506.0000      0.9639
43     11434                                    494.0000      0.9564
44     10035                                    470.0000      0.9914
45     11235                                    470.0000      0.9524
46     11432                                    468.0000      0.9713
47     10461                                    448.0000      0.9690
48     10301                                    445.0000      0.9329
49     10304                                    442.0000      0.9467
50     11224                                    440.0000      0.9920
51     11372                                    413.0000      0.9255
52     11435                                    412.0000      0.9518
53     11385                                    395.0000      0.8819
54     11214                                    387.0000      0.9685
55     10026                                    383.0000      0.9123
56     11373                                    370.0000      0.9432
57 

In [153]:
avg_per_zipcode = evi_svi_df['average_year_eviction_count_per_zipcode(z)'].mean()
avg_per_zipcode

np.float64(381.58045977011494)

In [152]:
average_svi = 0.80198
# https://www.atsdr.cdc.gov/place-health/php/svi/svi-interactive-map.html

In [155]:
evi_svi_df['above_eviction_average'] = evi_svi_df['average_year_eviction_count_per_zipcode(z)'] > avg_per_zipcode
evi_svi_df

zipcode  average_year_eviction_count_per_zipcode(z)  rpl_themes  \
0      10458                                   2221.0000      0.9891   
1      10467                                   2192.0000      0.9925   
2      10453                                   1994.0000      0.9983   
3      10456                                   1880.0000      0.9994   
4      10457                                   1765.0000      0.9989   
5      10468                                   1719.0000      0.9874   
6      11226                                   1639.0000      0.9220   
7      10452                                   1592.0000      0.9931   
8      10460                                   1557.0000      0.9954   
9      11212                                   1487.0000      0.9948   
10     11207                                   1320.0000      0.9839   
11     11233                                   1196.0000      0.9438   
12     11208                                   1129.0000      0.9581   
13     10466                                   1119.0000      0.9851   
14     10472                                   1070.0000      0.9937   
15     10459                                   1041.0000      0.9943   
16     11213                                   1011.0000      0.9805   
17     11203                                    952.0000      0.9386   
18     10451                                    932.0000      0.9897   
19     10462                                    922.0000      0.9708   
20     11221                                    897.0000      0.9352   
21     11210                                    843.0000      0.9174   
22     11368                                    810.0000      0.9633   
23     10463                                    746.0000      0.9862   
24     10029                                    724.0000      0.9788   
25     10032                                    689.0000      0.9771   
26     11225                                    682.0000      0.8538   
27     10031                                    678.0000      0.9323   
28     10455                                    664.0000      0.9971   
29     10033                                    627.0000      0.9650   
30     11236                                    612.0000      0.9117   
31     10469                                    602.0000      0.9507   
32     10034                                    579.0000      0.9541   
33     11216                                    559.0000      0.8297   
34     10454                                    546.0000      1.0000   
35     10030                                    546.0000      0.9828   
36     10473                                    544.0000      0.9868   
37     11206                                    543.0000      0.9679   
38     10027                                    532.0000      0.9312   
39     10040                                    513.0000      0.9857   
40     11230                                    513.0000      0.9702   
41     10039                                    506.0000      0.9748   
42     10037                                    506.0000      0.9639   
43     11434                                    494.0000      0.9564   
44     10035                                    470.0000      0.9914   
45     11235                                    470.0000      0.9524   
46     11432                                    468.0000      0.9713   
47     10461                                    448.0000      0.9690   
48     10301                                    445.0000      0.9329   
49     10304                                    442.0000      0.9467   
50     11224                                    440.0000      0.9920   
51     11372                                    413.0000      0.9255   
52     11435                                    412.0000      0.9518   
53     11385                                    395.0000      0.8819   
54     11214                                 

In [156]:
evi_svi_df['above_svi_average'] = evi_svi_df['rpl_themes'] > average_svi
evi_svi_df

zipcode  average_year_eviction_count_per_zipcode(z)  rpl_themes  \
0      10458                                   2221.0000      0.9891   
1      10467                                   2192.0000      0.9925   
2      10453                                   1994.0000      0.9983   
3      10456                                   1880.0000      0.9994   
4      10457                                   1765.0000      0.9989   
5      10468                                   1719.0000      0.9874   
6      11226                                   1639.0000      0.9220   
7      10452                                   1592.0000      0.9931   
8      10460                                   1557.0000      0.9954   
9      11212                                   1487.0000      0.9948   
10     11207                                   1320.0000      0.9839   
11     11233                                   1196.0000      0.9438   
12     11208                                   1129.0000      0.9581   
13     10466                                   1119.0000      0.9851   
14     10472                                   1070.0000      0.9937   
15     10459                                   1041.0000      0.9943   
16     11213                                   1011.0000      0.9805   
17     11203                                    952.0000      0.9386   
18     10451                                    932.0000      0.9897   
19     10462                                    922.0000      0.9708   
20     11221                                    897.0000      0.9352   
21     11210                                    843.0000      0.9174   
22     11368                                    810.0000      0.9633   
23     10463                                    746.0000      0.9862   
24     10029                                    724.0000      0.9788   
25     10032                                    689.0000      0.9771   
26     11225                                    682.0000      0.8538   
27     10031                                    678.0000      0.9323   
28     10455                                    664.0000      0.9971   
29     10033                                    627.0000      0.9650   
30     11236                                    612.0000      0.9117   
31     10469                                    602.0000      0.9507   
32     10034                                    579.0000      0.9541   
33     11216                                    559.0000      0.8297   
34     10454                                    546.0000      1.0000   
35     10030                                    546.0000      0.9828   
36     10473                                    544.0000      0.9868   
37     11206                                    543.0000      0.9679   
38     10027                                    532.0000      0.9312   
39     10040                                    513.0000      0.9857   
40     11230                                    513.0000      0.9702   
41     10039                                    506.0000      0.9748   
42     10037                                    506.0000      0.9639   
43     11434                                    494.0000      0.9564   
44     10035                                    470.0000      0.9914   
45     11235                                    470.0000      0.9524   
46     11432                                    468.0000      0.9713   
47     10461                                    448.0000      0.9690   
48     10301                                    445.0000      0.9329   
49     10304                                    442.0000      0.9467   
50     11224                                    440.0000      0.9920   
51     11372                                    413.0000      0.9255   
52     11435                                    412.0000      0.9518   
53     11385                                    395.0000      0.8819   
54     11214                                 

# **Step 4.3 Run Chi-Square test**
  **Null Hypothesis: There is no association between zipcodes that have above the average of black and hispanic percezipcodege and have above than average svi scores**

In [157]:
contingency_table = pd.crosstab(evi_svi_df.above_svi_average, evi_svi_df.above_eviction_average)
contingency_table

above_eviction_average  False  True 
above_svi_average                   
False                      50      0
True                       68     56

In [167]:
pd.set_option('display.float_format', lambda x: '%.2f' % x)

In [163]:
chi2, p_value, dof, expected = chi2_contingency(contingency_table)
chi2, p_value, dof, expected

(np.float64(31.25986599820354),
 np.float64(2.256975601346814e-08),
 1,
 array([[33.90804598, 16.09195402],
        [84.09195402, 39.90804598]]))

In [168]:
2.256975601346814e-08

2.256975601346814e-08

### **P-value is 0.0000 and chi-square is 31.25. We can reject the null hypothesis. The Chi-square suggests a strong-to-moderate deviation from the distribution if the two groups/categories are independent.**

# **conclusion: there is statistically significant association between the two "above average eviction" and "above average svi scores" zipcodes.**

In [169]:
a = evi_svi_df[(evi_svi_df['above_svi_average'] == True) & (evi_svi_df['above_eviction_average'] == True)].shape[0]
b = evi_svi_df[(evi_svi_df['above_svi_average'] == True) & (evi_svi_df['above_eviction_average'] == False)].shape[0]
c = evi_svi_df[(evi_svi_df['above_svi_average'] == False) & (evi_svi_df['above_eviction_average'] == True)].shape[0]
d = evi_svi_df[(evi_svi_df['above_svi_average'] == False) & (evi_svi_df['above_eviction_average'] == False)].shape[0]

In [170]:
observed = np.array([[a, b], [c, d]])
observed

array([[56, 68],
       [ 0, 50]])

In [171]:
a += 0.5
b += 0.5
c += 0.5
d += 0.5
# avoid 0 divisions

In [172]:
odds_ratio = (a * d) / (b * c)
odds_ratio
# extremely strong association between high svi and high evitction rates.

83.30656934306569

**This means, technically, zipcodes with above average svi had 83.31 times higher odds of having above average eviction rates compared to low-SVI zipcodes.** This high odds may be due to all high svi zipcodes had high evictions, and all high eviction zipcodes had high svi from the sample.

In [173]:
chi2, p_value, dof, expected = chi2_contingency(observed)
chi2, p_value

(np.float64(31.259865998203537), np.float64(2.256975601346814e-08))

# **Step 5: We also need a dataframe that has zipcodes as columns and black and hispanics percezipcodeges as rows and contents**

In [174]:
racial_avg_all_zp['black_hispanic_pct'] = racial_avg_all_zp['ep_afam'] + racial_avg_all_zp['ep_hisp']
racial_avg_all_zp

zipcode  evictions_per_zipcode_1kpop(z)  ep_afam  ep_asian  ep_hisp  \
0      10458                           26.86    16.10      3.30    71.30   
1      10037                           26.44    63.10      3.80    19.80   
2      10474                           26.33    26.40      0.20    70.20   
3      10460                           26.01    26.50      1.40    65.00   
4      10453                           24.81    24.60      1.60    69.20   
5      10467                           22.21    28.60      6.20    52.90   
6      10457                           22.11    30.90      0.40    65.30   
7      10456                           21.22    38.20      0.70    56.30   
8      10468                           21.12    15.60      2.30    78.00   
9      10452                           20.25    28.10      1.00    65.40   
10     10459                           20.03    28.90      0.20    67.30   
11     10451                           18.16    37.20      1.40    55.00   
12     11212                           17.70    70.10      0.80    19.50   
13     10030                           17.46    53.80      3.60    29.70   
14     10470                           17.35    31.30      2.00    22.80   
15     10039                           16.93    54.70      1.90    32.70   
16     10472                           16.39    23.10      8.20    61.70   
17     11226                           16.22    63.20      3.20    14.90   
18     10466                           15.48    59.80      2.60    29.40   
19     10455                           14.96    21.10      1.10    74.10   
20     11213                           14.62    56.60      1.60    14.30   
21     11233                           14.39    66.70      1.30    15.70   
22     10034                           13.93     5.70      3.30    69.00   
23     11207                           13.64    55.90      1.50    32.80   
24     10454                           13.29    24.60      0.20    70.30   
25     11210                           13.29    51.60      5.80     7.50   
26     11239                           13.15    61.70      2.60    15.90   
27     10035                           12.18    36.70      3.40    43.00   
28     11203                           12.13    80.30      1.80     6.90   
29     10462                           11.94    22.20     16.60    47.20   
30     10040                           11.75     2.90      1.80    71.80   
31     11225                           11.66    53.70      3.30    10.80   
32     10032                           11.39    12.80      4.50    65.70   
33     10031                           11.11    23.60      4.00    50.90   
34     10301                           11.03    19.90      7.60    26.30   
35     10033                           10.68     3.60      3.30    67.20   
36     10463                           10.61    12.10      3.70    52.30   
37     11208                           10.44    45.80      8.10    39.00   
38     10026                           10.01    47.60      6.30    19.80   
39     10304                            9.85    26.00     12.30    21.70   
40     11221                            9.83    39.70      5.10    31.00   
41     10029                            9.57    25.00     10.80    45.70   
42     11224                            9.19    20.10      9.50    19.50   
43     10473                            9.05    30.90      2.10    62.20   
44     11216                            8.96    44.10      4.60    13.60   
45     10461                            8.81     7.70     12.80    49.70   
46     11415                            8.76     6.70     22.30    22.90   
47     10469                            8.38    52.20      5.50    29.50   
48     11433                            8.36    55.30     15.40    13.90   
49     10303                            8.23    30.00      9.80    39.00   
50     11423                            8.15    33.40     30.60    14.70   
51     10019                            8.09     4.50

In [177]:
racial_avg_all_zp.columns

Index(['zipcode', 'evictions_per_zipcode_1kpop(z)', 'ep_afam', 'ep_asian',
       'ep_hisp', 'ep_nhpi', 'ep_white', 'ep_twomore', 'ep_otherrace',
       'rpl_themes', 'black_hispanic_pct'],
      dtype='object')

In [178]:
evi_bh_df = racial_avg_all_zp[['zipcode', 'evictions_per_zipcode_1kpop(z)', 'black_hispanic_pct']]
evi_bh_df

zipcode  evictions_per_zipcode_1kpop(z)  black_hispanic_pct
0      10458                           26.86               87.40
1      10037                           26.44               82.90
2      10474                           26.33               96.60
3      10460                           26.01               91.50
4      10453                           24.81               93.80
5      10467                           22.21               81.50
6      10457                           22.11               96.20
7      10456                           21.22               94.50
8      10468                           21.12               93.60
9      10452                           20.25               93.50
10     10459                           20.03               96.20
11     10451                           18.16               92.20
12     11212                           17.70               89.60
13     10030                           17.46               83.50
14     10470                           17.35               54.10
15     10039                           16.93               87.40
16     10472                           16.39               84.80
17     11226                           16.22               78.10
18     10466                           15.48               89.20
19     10455                           14.96               95.20
20     11213                           14.62               70.90
21     11233                           14.39               82.40
22     10034                           13.93               74.70
23     11207                           13.64               88.70
24     10454                           13.29               94.90
25     11210                           13.29               59.10
26     11239                           13.15               77.60
27     10035                           12.18               79.70
28     11203                           12.13               87.20
29     10462                           11.94               69.40
30     10040                           11.75               74.70
31     11225                           11.66               64.50
32     10032                           11.39               78.50
33     10031                           11.11               74.50
34     10301                           11.03               46.20
35     10033                           10.68               70.80
36     10463                           10.61               64.40
37     11208                           10.44               84.80
38     10026                           10.01               67.40
39     10304                            9.85               47.70
40     11221                            9.83               70.70
41     10029                            9.57               70.70
42     11224                            9.19               39.60
43     10473                            9.05               93.10
44     11216                            8.96               57.70
45     10461                            8.81               57.40
46     11415                            8.76               29.60
47     10469                            8.38               81.70
48     11433                            8.36               69.20
49     10303                            8.23               69.00
50     11423                            8.15               48.10
51     10019                            8.09               21.40
52     10027                            8.08               59.00
53     10036                            7.82               27.20
54     10018                            7.39               19.50
55     11432                            7.28               37.80
56     11434                            7.25               89.70
57     11368                            7.18               83.30
58     10475                            7.01               90.60
59     11435                            6.87               53.60
60     10471                            

In [179]:
average_bh_pct = evi_bh_df['black_hispanic_pct'].mean()
average_bh_pct

np.float64(44.84022988505747)

In [180]:
avg_per_zipcode, average_bh_pct
# evictions and black + hispanic pct

(np.float64(381.58045977011494), np.float64(44.84022988505747))

In [188]:
above_evi_1kpop_average = evi_bh_df['evictions_per_zipcode_1kpop(z)'].mean()
above_evi_1kpop_average

np.float64(6.958997919538344)

In [191]:
evi_bh_df['above_evi_average'] = evi_bh_df['evictions_per_zipcode_1kpop(z)'] > above_evi_1kpop_average
evi_bh_df

zipcode  evictions_per_zipcode_1kpop(z)  black_hispanic_pct  \
0      10458                           26.86               87.40   
1      10037                           26.44               82.90   
2      10474                           26.33               96.60   
3      10460                           26.01               91.50   
4      10453                           24.81               93.80   
5      10467                           22.21               81.50   
6      10457                           22.11               96.20   
7      10456                           21.22               94.50   
8      10468                           21.12               93.60   
9      10452                           20.25               93.50   
10     10459                           20.03               96.20   
11     10451                           18.16               92.20   
12     11212                           17.70               89.60   
13     10030                           17.46               83.50   
14     10470                           17.35               54.10   
15     10039                           16.93               87.40   
16     10472                           16.39               84.80   
17     11226                           16.22               78.10   
18     10466                           15.48               89.20   
19     10455                           14.96               95.20   
20     11213                           14.62               70.90   
21     11233                           14.39               82.40   
22     10034                           13.93               74.70   
23     11207                           13.64               88.70   
24     10454                           13.29               94.90   
25     11210                           13.29               59.10   
26     11239                           13.15               77.60   
27     10035                           12.18               79.70   
28     11203                           12.13               87.20   
29     10462                           11.94               69.40   
30     10040                           11.75               74.70   
31     11225                           11.66               64.50   
32     10032                           11.39               78.50   
33     10031                           11.11               74.50   
34     10301                           11.03               46.20   
35     10033                           10.68               70.80   
36     10463                           10.61               64.40   
37     11208                           10.44               84.80   
38     10026                           10.01               67.40   
39     10304                            9.85               47.70   
40     11221                            9.83               70.70   
41     10029                            9.57               70.70   
42     11224                            9.19               39.60   
43     10473                            9.05               93.10   
44     11216                            8.96               57.70   
45     10461                            8.81               57.40   
46     11415                            8.76               29.60   
47     10469                            8.38               81.70   
48     11433                            8.36               69.20   
49     10303                            8.23               69.00   
50     11423                            8.15               48.10   
51     10019                            8.09               21.40   
52     10027                            8.08               59.00   
53     10036                            7.82               27.20   
54     10018                            7.39               19.50   
55     11432                            7.28               37.80   
56     11434                            7.25               89.70   
57     11368                            7.18               83

In [192]:
evi_bh_df['above_bh_average'] = evi_bh_df['black_hispanic_pct'] > average_bh_pct
evi_bh_df

zipcode  evictions_per_zipcode_1kpop(z)  black_hispanic_pct  \
0      10458                           26.86               87.40   
1      10037                           26.44               82.90   
2      10474                           26.33               96.60   
3      10460                           26.01               91.50   
4      10453                           24.81               93.80   
5      10467                           22.21               81.50   
6      10457                           22.11               96.20   
7      10456                           21.22               94.50   
8      10468                           21.12               93.60   
9      10452                           20.25               93.50   
10     10459                           20.03               96.20   
11     10451                           18.16               92.20   
12     11212                           17.70               89.60   
13     10030                           17.46               83.50   
14     10470                           17.35               54.10   
15     10039                           16.93               87.40   
16     10472                           16.39               84.80   
17     11226                           16.22               78.10   
18     10466                           15.48               89.20   
19     10455                           14.96               95.20   
20     11213                           14.62               70.90   
21     11233                           14.39               82.40   
22     10034                           13.93               74.70   
23     11207                           13.64               88.70   
24     10454                           13.29               94.90   
25     11210                           13.29               59.10   
26     11239                           13.15               77.60   
27     10035                           12.18               79.70   
28     11203                           12.13               87.20   
29     10462                           11.94               69.40   
30     10040                           11.75               74.70   
31     11225                           11.66               64.50   
32     10032                           11.39               78.50   
33     10031                           11.11               74.50   
34     10301                           11.03               46.20   
35     10033                           10.68               70.80   
36     10463                           10.61               64.40   
37     11208                           10.44               84.80   
38     10026                           10.01               67.40   
39     10304                            9.85               47.70   
40     11221                            9.83               70.70   
41     10029                            9.57               70.70   
42     11224                            9.19               39.60   
43     10473                            9.05               93.10   
44     11216                            8.96               57.70   
45     10461                            8.81               57.40   
46     11415                            8.76               29.60   
47     10469                            8.38               81.70   
48     11433                            8.36               69.20   
49     10303                            8.23               69.00   
50     11423                            8.15               48.10   
51     10019                            8.09               21.40   
52     10027                            8.08               59.00   
53     10036                            7.82               27.20   
54     10018                            7.39               19.50   
55     11432                            7.28               37.80   
56     11434                            7.25               89.70   
57     11368                            7.18               83

### **This is the zipcodes evictions + black/hispanic chi-test df:**

# **Run Chi-Square test**
  **Null Hypothesis: There is no association between zipcodes that have above the average of black and hispanic per zipcodes and have above than average svi scores**

In [193]:
contingency_table = pd.crosstab(evi_bh_df.above_bh_average, evi_bh_df.above_evi_average)
contingency_table

above_evi_average  False  True 
above_bh_average               
False                 91      6
True                  24     53

In [194]:
chi2, p_value, dof, expected = chi2_contingency(contingency_table)
chi2, p_value, dof, expected

(np.float64(72.40035499223367),
 np.float64(1.7568234771276802e-17),
 1,
 array([[64.1091954, 32.8908046],
        [50.8908046, 26.1091954]]))

In [195]:
p_value

np.float64(1.7568234771276802e-17)

### **P-value is 0.0000 (extremely small) and chi-squiare is 72.40. We can refute the null hypothesis. We can say we are confident that high "black+hispanic percetage" boroughs are more likely to have high evictions (or vice versa) or there is a strong association between high BH pct and high eviction rates in the full dataset**

# **conclusion: there is a statistically significant association between the two "above average eviction" and "above average black+hispanic pct" zipcodes.**

In [196]:
a = evi_bh_df[(evi_bh_df['above_bh_average'] == True) & (evi_bh_df['above_evi_average'] == True)].shape[0]
b = evi_bh_df[(evi_bh_df['above_bh_average'] == True) & (evi_bh_df['above_evi_average'] == False)].shape[0]
c = evi_bh_df[(evi_bh_df['above_bh_average'] == False) & (evi_bh_df['above_evi_average'] == True)].shape[0]
d = evi_bh_df[(evi_bh_df['above_bh_average'] == False) & (evi_bh_df['above_evi_average'] == False)].shape[0]

In [197]:
observed = np.array([[a, b], [c, d]])
observed

array([[53, 24],
       [ 6, 91]])

In [198]:
odds_ratio = (a * d) / (b * c)
odds_ratio

33.49305555555556

**This means, technically, zipcodes with above average b+h pct had 33.4931 times higher odds of having above average eviction rates compared to low-bh-pct zipcodes in nyc.** This high odds may be due to all high bh pct zipcodes had high evictions, and all high eviction zipcodes had high bh pct from the sample.

In [200]:
chi2, p_value, dof, expected = chi2_contingency(observed)
chi2, p_value
# 72.400 is a strong deviation from expected undert the null hypothesis

(np.float64(72.40035499223367), np.float64(1.7568234771276802e-17))